File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJFilesByNameIndex.java
package org.cappuccino_project.ide.intellij.plugin.indices;

import com.intellij.psi.stubs.StubIndexKey;
import org.cappuccino_project.ide.intellij.plugin.lang.ObjJFile;
import org.jetbrains.annotations.NotNull;

public class ObjJFilesByNameIndex extends ObjJStringStubIndexBase<ObjJFile> {

    private static final ObjJFilesByNameIndex INSTANCE = new ObjJFilesByNameIndex();

    private static final StubIndexKey<String, ObjJFile> KEY = IndexKeyUtil.createIndexKey(ObjJFilesByNameIndex.class);

    private static final int VERSION = 3;


    private ObjJFilesByNameIndex(){}

    public int getVersion() {
        return super.getVersion() + VERSION;
    }

    @NotNull
    public static ObjJFilesByNameIndex getInstance() {
        return INSTANCE;
    }

    @NotNull
    @Override
    protected Class<ObjJFile> getIndexedElementClass() {
        return ObjJFile.class;
    }

    @NotNull
    @Override
    public StubIndexKey<String, ObjJFile> getKey() {
        return KEY;
    }


}

File: src/cappuccino/ide/intellij/plugin/indices/ObjJFilesByNameIndex.kt
package cappuccino.ide.intellij.plugin.indices

import com.intellij.psi.stubs.StubIndexKey
import cappuccino.ide.intellij.plugin.lang.ObjJFile

class ObjJFilesByNameIndex private constructor() : ObjJStringStubIndexBase<ObjJFile>() {

    override val indexedElementClass: Class<ObjJFile>
        get() = ObjJFile::class.java

    override fun getVersion(): Int {
        return super.getVersion() + VERSION
    }

    override fun getKey(): StubIndexKey<String, ObjJFile> {
        return KEY
    }

    companion object {

        val instance = ObjJFilesByNameIndex()

        private val KEY = IndexKeyUtil.createIndexKey(ObjJFilesByNameIndex::class.java)

        private const val VERSION = 3
    }


}
